pkgs/tools/text/validator-nu/default.nix

```
@@ -40,29 +40,44 @@ let
   };
 
 in
-stdenvNoCC.mkDerivation rec {
+stdenvNoCC.mkDerivation (finalAttrs: {
   inherit pname version src;
 
   nativeBuildInputs = [ git jdk_headless makeWrapper python3 ];
 
+  postPatch = ''
+    substituteInPlace build/build.py --replace-warn \
+      'validatorVersion = "%s.%s.%s" % (year, month, day)' \
+      'validatorVersion = "${finalAttrs.version}"'
+  '';
+
   buildPhase = ''
     ln -s '${deps}/dependencies' '${deps}/extras' .
     JAVA_HOME='${jdk_headless}' python checker.py build
   '';
 
   installPhase = ''
+    runHook preInstall
+
     mkdir -p "$out/bin" "$out/share/java"
     mv build/dist/vnu.jar "$out/share/java/"
     makeWrapper "${jre_headless}/bin/java" "$out/bin/vnu" \
       --add-flags "-jar '$out/share/java/vnu.jar'"
+
+    runHook postInstall
+  '';
+
+  doInstallCheck = true;
+  installCheckPhase = ''
+    $out/bin/vnu --version | grep ${finalAttrs.version}
```

The version number contains `.`, which matches any character when reinterpreted as a regex.

```suggestion
    "$out/bin/vnu" --version | grep -Fx '${finalAttrs.version}'
```

---

Thanks for the review and suggestion, @andersk, great catch.
What's the motivation for quoting the binary path? Is there a possibility `$out` contains characters interpreted by the shell if not escaped, e.g. blank space? I'd expect it to be something along the lines of `/nix/store/*validator-nu`, that said I'm wondering how nix behaves when doing a `nix build` and the current working directory contains a blank space 🤔

---

Using correct shell quoting everywhere is simpler, easier, and more correct than trying to reason about exactly where it’s strictly necessary. I wouldn’t complain about it in other people’s nixpkgs code if it were my only comment, but yeah, in my own suggestion for my own package I’m going to hold myself to that standard.

---

Thanks for the added context, @andersk, I've made the changes you suggested and threw in grep's `-q` option for good measure…